gluten-core/src/main/scala/org/apache/gluten/utils/PullOutProjectHelper.scala

```diff
@@ -61,7 +61,13 @@ trait PullOutProjectHelper {
       replaceBoundReference: Boolean = false): Expression =
     expr match {
       case alias: Alias =>
-        projectExprsMap.getOrElseUpdate(alias.child.canonicalized, alias).toAttribute
+        alias.child match {
+          case _: Literal =>
+            projectExprsMap.getOrElseUpdate(alias.child.canonicalized, alias)
```

Comment:
Does `getMissingLiterals` method do the same things with this code ? This code and `getMissingLiterals` say if there is an `Alias(Literal)` then we add one more `Alias(Literal)` to project. What happens if there are three `Alias(Literal)` ?

Do you mean we should always include `Alias(Literal)` ? if so we can change `projectExprsMap.getOrElseUpdate(alias.child.canonicalized, alias)` to `projectExprsMap.getOrElseUpdate(alias, alias)` and remove `getMissingLiterals`.

Reply:
Yes, all literal alias should be added.